app/models/katello/glue/pulp/repo.rb

```diff
@@ -41,20 +41,20 @@ def self.distribution_bootable?(distribution)
           end
         end
 
-        def self.needs_importer_updates(repos, capsule_content)
+        def self.needs_importer_updates(repos, smart_proxy_service)
           repos.select do |repo|
-            repo_details = capsule_content.pulp_repo_facts(repo.pulp_id)
+            repo_details = smart_proxy_service.pulp_repo_facts(repo.pulp_id)
```

Comment:
Going through the repository backend service would be much better to get these details:

repo.backend_service(smart_proxy).backend_data

Then these methods don't even need to know about the service.